### ISSUE TITLE
USER ISSUE: Connection Lost over and over

### ISSUE THREAD

**Issue by dphotograph** [StrangeLoopGames/EcoIssues#11068]
**Version:** 0.7.8.9 beta

**Steps to Reproduce:**
i try to connect to an new Server that i rent on Nitrado.net.
i see the server, i can connect, the Password required box apeares i wrote mine in and connect sometimes i get seconds on the server i walk 4 steps and the connection lost again 
 

**Expected behavior:**
connections are possible 

**Actual behavior:**
The following error message popped up:
Connection times out
Connection timed out
```

**Do you have mods installed? Does issue happen when no mods are installed?:**
The following error message popped up:
Connection times out
Connection timed out
```

**Comment by johnkslg** [comment]
Hopefully fixed, can work with user as a staging testcase

**Comment by metachronism** [comment]
@mirasrael please investigate more, likely not relevant on 8.0 now though with all your other changes

**Comment by dphotograph** [comment]
If I can provide you with additional information such as logs, server access information or other helpful data, please let me know.

**Comment by mirasrael** [comment]
@dphotograph Hi, do you still experiencing this problem after 8.0 release?

**Comment by dphotograph** [comment]
@mirasrael Hi, after my Server was updated to 8.0 i try it but the problem is still the same with one exeption, now i lost the connection on the World Loading. it Still see the same error.  (sorry for the bad english)

![image](https://user-images.githubusercontent.com/9300651/52521124-8aea1000-2c72-11e9-84f6-0254114eb08d.png)

**Comment by mirasrael** [comment]
@dphotograph any improvements after 8.0.1?

**Comment by dphotograph** [comment]
Hello @mirasrael 
sorry for the delay time i was realy busy the days. 
I Just reinstall Eco on my server and try 0.8.0.3 beta on booth but nothing changed i think.

**Comment by dphotograph** [comment]
Connection timed out

Log File:
Initialize engine version: 2018.2.20f1 (cef3e6c0c622)
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1080 Ti (ID=0x1b06)
    Vendor:   
    VRAM:     11127 MB
    Driver:   25.21.14.1771
<RI> Initializing input.

<RI> Input initialized.

<RI> Initialized touch support.

UnloadTime: 1.163227 ms
Client version: 0.8.0.3 beta
Thursday, February 14, 2019 5:45 PM
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Settings:Awake()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<ReceiveModData>c__Iterator2:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<ReceiveModData>c__Iterator2:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

**Comment by dphotograph** [comment]
after an complete reinstall of my Client the Log had changed a little bit  (deinstall Eco, deleted all files from the %appdata% and install Eco new over Steam

**Comment by dphotograph** [comment]
Connection timed out

Log File:
Initialize engine version: 2018.2.20f1 (cef3e6c0c622)
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1080 Ti (ID=0x1b06)
    Vendor:   
    VRAM:     11127 MB
    Driver:   25.21.14.1771
<RI> Initializing input.

<RI> Input initialized.

<RI> Initialized touch support.

UnloadTime: 0.849733 ms
Client version: 0.8.0.3 beta
Thursday, February 14, 2019 5:45 PM
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Settings:Awake()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<ReceiveModData>c__Iterator2:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

**Comment by dphotograph** [comment]
Connection timed out

Log File:
Initialize engine version: 2018.2.20f1 (cef3e6c0c622)
GfxDevice: creating device client; threaded=1
Direct3D:
    Version:  Direct3D 11.0 [level 11.1]
    Renderer: NVIDIA GeForce GTX 1080 Ti (ID=0x1b06)
    Vendor:   
    VRAM:     11127 MB
    Driver:   25.21.14.1771
<RI> Initializing input.

<RI> Input initialized.

<RI> Initialized touch support.

UnloadTime: 0.432491 ms
Client version: 0.8.0.3 beta
Thursday, February 14, 2019 5:45 PM
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Settings:Awake()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Failed to send packet: System.Net.Sockets.SocketException (0x80004005): Error looking up error string
  at System.Net.Sockets.Socket.SendTo (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socketFlags, System.Net.EndPoint remoteEP) [0x00000] in <00000000000000000000000000000000>:0 
  at Lidgren.Network.NetPeer.SendPacket (System.Int32 numBytes, System.Net.IPEndPoint target, System.Int32 numMessages, System.Boolean& connectionReset) [0x00000] in <00000000000000000000000000000000>:0 
  at Lidgren.Network.NetPeer.Heartbeat () [0x00000] in <00000000000000000000000000000000>:0 
  at Lidgren.Network.NetPeer.NetworkLoop () [0x00000] in <00000000000000000000000000000000>:0 
  at RootMotion.FinalIK.Grounder+GrounderDelegate.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00000] in <00000000000000000000000000000000>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x00000] in <00000000000000000000000000000000>:0 
  at RootMotion.FinalIK.Grounder+GrounderDelegate.Invoke () [0x00000] in <00000000000000000000000000000000>:0 
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:Log(Object)
Eco.Shared.Utils.Log:WriteError(LocString)
Eco.Shared.Networking.NetworkClient:HandleIncomingMessage(NetIncomingMessage)
Eco.Shared.Networking.NetworkClient:<Initialize>b__42_0()
RootMotion.FinalIK.GrounderDelegate:Invoke()
System.Threading.ExecutionContext:RunInternal(ExecutionContext, ContextCallback, Object, Boolean)
System.Threading.ExecutionContext:Run(ExecutionContext, ContextCallback, Object)
System.Threading.ThreadHelper:ThreadStart()
RootMotion.FinalIK.GrounderDelegate:Invoke()
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following icons: CeibaSeedPackItem, FarmingFocusedWorkflowTalentGroup, FarmingFrugalWorkspaceTalentGroup, FarmingLavishWorkspaceTalentGroup, FarmingParallelProcessingTalentGroup, SaguaroSeedPackItem
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<CreateMissingItems>c__Iterator3:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<BuildItems>c__Iterator4:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

Missing following objects: APGenObject, ButtonObject, ElkSkypiercerObject, LogicGateObject, SwitchObject, TollObject
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
<BuildFromModKit>c__Iterator0:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
<ReceiveModData>c__Iterator2:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
 
(Filename: C:\buildslave\unity\build\Runtime/Export/Debug.bindings.h Line: 43)

**Comment by D3nnis3n** [comment]
Fixed in 8.0.5, reopen if reoccurs.
